pandas/core/series.py

```diff
@@ -4056,7 +4056,9 @@ def apply(
             Python function or NumPy ufunc to apply.
         convert_dtype : bool, default True
             Try to find better dtype for elementwise function results. If
-            False, leave as dtype=object.
+            False, leave as dtype=object. For the dtypes Categorical,
+            Sparse, Interval, Period, DatetimeArray and TimedeltaArray
+            the original dtype is kept.
```

Comment:
Hi @sukriti1 - how did you find this list of types? Was it from the code, or are they documented elsewhere?

Reply:
Hi @MarcoGorelli - I found this list of type from this comment :
https://github.com/pandas-dev/pandas/issues/39580#issuecomment-774973540

Reply:
I see, thanks!

OK, so instead of listing the dtypes here (which could expand in the future) maybe it's better to just say that the conversion doesn't happen for extension array dtypes, linking to the docs for `is_extension_array_dtype`?

cc @attack68

Reply:
@MarcoGorelli Changes made!

Reply:
Agreeing with @MarcoGorelli's second thought here, I do like the original wording better. Do we have tests for this?

Reply:
Perhaps we can also add something to the effect that the list is not necessarily exhaustive.

Reply:
Like, "For some dtypes, such as Categorical,
            Sparse, Interval, Period, DatetimeArray, and TimedeltaArray,
            the original dtype is kept."?

Reply:
I am in favor of explicitly listing the dtypes. Most users don't know what an extension array type is. I have been using pandas quite regularly for years and never heard of extension arrays.